Add img mapper

maggiemay-afk opened the pull request:
Added react image mapper to change game from multiple choice style questions to users selecting a map area when making a guess at what zone the soundtrack is from. Also, major cleanup of unused code, components, soundtracks, etc. and styling changes to accommodate new game play style.  

- Add images (maps) for all WoW zones
- map details / clickable areas for all zones in an images
- Use stack data type for state to handle clicking on map zones and using the "back one map" button
- Add zone name to image on mouse hover of a clickable map area
- Clean up of config files to remove livestreams and other unusable soundtracks
- Clean up of unused, commented out code
- Remove unused components (i.e., Background Image)
- Remove volume slider 
- Redo styling of home page, such as not using full-screen for YouTube videos

maggiemay-afk committed:
Started work on image mapper

maggiemay-afk committed:
add package and yarn lock

maggiemay-afk committed:
more map-details, add stack for state

maggiemay-afk committed:
update map state to stack

maggiemay-afk committed:
evaluate user choice, refactor getQuestion and YT player

maggiemay-afk committed:
Merge branch 'main' into add_img_mapper

maggiemay-afk committed:
Working on scoring, styling

maggiemay-afk committed:
Merge branch 'add_img_mapper' of github.com:hermanator608/wow-soundtrack-guesser into add_img_mapper

maggiemay-afk committed:
updated styling of youtube video, etc.

maggiemay-afk committed:
added more image maps, fixed issue with back button

maggiemay-afk committed:
finish all maps, update videos

maggiemay-afk committed:
Merge branch 'main' into add_img_mapper

maggiemay-afk committed:
Clean up of unused components, code, styles, etc.

maggiemay-afk committed:
Missed a ==, updated to ===

maggiemay-afk commented:
Sorry, @hermanator608 - kind of a hefty PR. Let me know if you want me to break it into a couple smaller ones. 
Thank you!

hermanator608 commented on src/App.css:
```diff
@@ -37,19 +38,68 @@
   overflow: hidden;
   padding: 30px;
   padding-bottom: calc(30px + env(safe-area-inset-bottom));
+} */
```
Delete this if not using?

hermanator608 commented on src/App.tsx:
```diff
@@ -1,63 +1,145 @@
 
 
-function App() {
+function App(props: any) {
```
Did you have to add props here? Looks like they are not used?

maggiemay-afk replied:
Added this by mistake. No, don't need any props here

hermanator608 commented on src/App.tsx:
```diff
@@ -1,63 +1,145 @@
+    if (!area.id) {
+      alert("Bug - id not defined")
+      return;
+    }
```
Do we really want to alert to a user if we dont have this defined?

hermanator608 commented on src/App.tsx:
```diff
@@ -1,63 +1,145 @@
-  };
+  }
+
+  //TODO: Change from alert to something else
```
Should we finish this TODO? Or do you plan to update in followup PR?

maggiemay-afk replied:
Plan to update this in a follow up PR - will be adding scoring, and this would be part of the scoring functionality.

hermanator608 commented on src/App.tsx:
```diff
@@ -1,63 +1,145 @@
+  }
+
+  const displayFeedback = (result: Boolean) => {
+    setResult(result);
+    if (result) setScore(score + 1);
+    // if (score === 5) //TODO: End Game
```
Looks like we set score but dont display it anywhere?

maggiemay-afk replied:
Removing this line as well as the state for score. Similar to comment above, will be doing a follow up PR for scoring.

maggiemay-afk replied:
More specifically, removing line 81, 82, and 24 in the App.tsx file

hermanator608 commented on src/config/types.ts:
```diff
@@ -8,3 +9,82 @@ export type Ambiance = {
+  getData(): Array<MapDetails>
+}
+
+export class Stack<MapDetails> implements stackInterface<MapDetails> {
```
Can we move this to something like - `src/utils/MapStack.ts`?

hermanator608 replied:
Please update this too :ty:

hermanator608 commented on src/config/wow/miscellaneous.ts:
```diff
@@ -84,47 +32,27 @@ export const miscellaneous = [
   {
-    name: 'Azuremyst Isle - Gameplay',
+    name: 'Azuremyst Isle',
     code: 'PmWAbDdP4mY',
     group: 'Gameplay',
     channel: channels.Meisio,
   },
   {
-    name: 'Elwynn Forest - Gameplay',
+    name: 'Elwynn Forest',
     code: 'UURb7kXLz0A',
     group: 'Gameplay',
     channel: channels.Meisio,
   },
   {
-    name: 'Dun Morogh - Gameplay',
+    name: 'Dun Morogh',
     code: 'TG5U-tlRLWw',
     group: 'Gameplay',
     channel: channels.Meisio,
   },
   {
-    name: 'Mulgore - Gameplay',
+    name: 'Mulgore',
     code: 'hJba3NVLtz8',
     group: 'Gameplay',
     channel: channels.Meisio,
```
Might just want to get rid of all these "gameplay" ones at they dont focus on the music and have other distracting noises like fighting, quests pickup/turn-in, etc

hermanator608 replied:
bump

hermanator608 commented on src/state.ts:
```diff
@@ -4,26 +4,32 @@ import { worldOfWarcraft } from "./config";
-  answer: number;
-  options: number[];
+  //answer: number;
+  //options: number[];
```
We should remove these if not used

hermanator608 commented on src/state.ts:
```diff
@@ -4,26 +4,32 @@ import { worldOfWarcraft } from "./config";
-  const answer = Math.floor(Math.random() * options.length);
-
-  console.log(options, options[answer]);
+  //const options: number[] = [];
+  // while (options.length < 5) {
+  //   const randomNum = Math.floor(Math.random() * worldOfWarcraft.length);
+  //   if (!options.includes(randomNum)) {
+  //     options.push(randomNum);
+  //   }
+  // }
+  //const answer = Math.floor(Math.random() * options.length);
```
Remove?

hermanator608 commented on src/state.ts:
```diff
@@ -4,26 +4,32 @@ import { worldOfWarcraft } from "./config";
-    answer: options[answer],
-    options,
+    //answer: options[answer],
+    //options,
```
rm

hermanator608 commented on src/state.ts:
```diff
@@ -4,26 +4,32 @@ import { worldOfWarcraft } from "./config";
+
+  const answerIndex = Math.floor(Math.random() * worldOfWarcraft.length);
+  let answerName = worldOfWarcraft[answerIndex].name;
+  console.log(answerName);
```
Probably should remove this? Otherwise users could cheat?

hermanator608 commented on src/App.tsx:
```diff
@@ -1,63 +1,145 @@
+          variant="contained" 
+          startIcon={<ReplayIcon />}
+        >
+        Back One Map
```
Maybe we can just update this to `back`? I think it'd be better to move this button to the top left of the map image. The location for me is a bit confusing.

hermanator608 replied:
Or even could just be a back arrow in the top left corner of the Map Image?

hermanator608 reviewed (changes requested):
Great work so far! I'm running into a few issues with the deployed application - 

* The first video will not start - does not work until I just select a random zone and fail and then the next video starts playing sound as it should
* The pause, skip, restart buttons are not working


Also seeing the following console error log:
```
api.ts:209 Uncaught FirebaseError: Firebase: No Firebase App '[DEFAULT]' has been created - call initializeApp() first (app/no-app).
    at wt (api.ts:209:25)
    at lr (api.ts:73:49)
    at logEventClickWrapper.ts:15:21
    at logEventClickWrapper.ts:23:5
    at Object.Le (react-dom.production.min.js:54:317)
    at je (react-dom.production.min.js:54:471)
    at react-dom.production.min.js:55:35
    at Fr (react-dom.production.min.js:105:68)
    at zr (react-dom.production.min.js:106:380)
    at react-dom.production.min.js:117:104
```

maggiemay-afk committed:
Correctly initialize firebase, update back button, remove unused code

maggiemay-afk commented:
Recap of new changes: 

- Correctly initialize firebase
(this change also seemed to fix the issue with buttons not working. Buttons appear to be working now on the GitHub test URL.
- Deleted the commented out code, unused code, and logs used for testing 
- Updated the back button, overlaying this on the image now
- Removed the bits of code pertaining to scoring. Will be doing a new PR just for scoring

maggiemay-afk committed:
don't alert the user, log the error if a map area id is undefined

hermanator608 commented on src/config/types.ts:
```diff
@@ -8,3 +9,82 @@ export type Ambiance = {
+  areas: Array<MapAreas>
+}
+
+export interface stackInterface<MapDetails> {
```
Nit: Capitalize this please

maggiemay-afk committed:
move Stack class, remove gameplay videos

maggiemay-afk commented:
Sorry - missed those comments you had made earlier 
Changes: 
- Remove gameplay videos from the miscellaneous.ts file
- Move the Stack class out of types.ts and into new file, src/utils/MapStack.ts
- Update StackInterface - capital

hermanator608 reviewed (approved):
LGTM great work! 🔥
